### PR TITLE
Improve "all time" warning UI

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

- Follow-up to #7849. Improves the UI by only showing a warning when it's relevant for customers (i.e. if you have an event from before 2015).

## How did you test this code?
- Banner is not shown when selecting "All time"
- Sent an event from before 2015 and banner is now shown.

### CODE REVIEW

File: posthog/api/team.py [hunk 1]
```diff
@@ -90,6 +92,18 @@ def get_effective_membership_level(self, team: Team) -> Optional[OrganizationMem
...
         return GroupTypeMapping.objects.filter(team=team).exists()
 
+    def get_event_first_seen(self, team: Team) -> Optional[datetime.datetime]:
+        # TODO: Remove try statement when CH is moved out ee/
+        try:
+            from ee.clickhouse.client import sync_execute
+        except ImportError:
+            return None
+
+        response = sync_execute(
```
Comment (paolodamico) [inline posthog/api/team.py:102]: Seems like an innocuous enough query as a) it's very light, and b) we only use this serializer when retrieving the current project (i.e. `/api/projects/{id}`) not on project list or nested serializers, but open to challenge on whether this should be computed async.

Comment (macobo) [reply posthog/api/team.py]: Need to think on this - we already are doing a lot of unneeded clickhouse load every pageload, and this smells like the same. Will think on this.

Comment (paolodamico) [reply posthog/api/team.py]: For Q2, it shouldn't. If CH is down, we will just return `None` which should be interpreted as unable to deetermine.

**Comment by macobo** [comment]
Suggestion: Let's have a separate endpoint for this data, hit it only when "all of time" is selected. Only load it once (no need to reset every time)

**Review by clarkus — COMMENTED** [review]
Does this need to need to be shown every time the user selects "all time". Should it be dismissible? It displaces the chart under the controls, so it might be kind of annoying if it's always shown. If our goal is getting users to learn what "all time" means, we might try some overlay component that has higher priority, displaces the related content less, and lends well to being dismissed once the user acknowledges it.

**Comment by paolodamico** [comment]
## Original scope of this PR
Alright, in terms of the original scope of this PR:
- @macobo, I've updated so we now have a separate endpoint for this (please read more details below).
- @clarkus I opted to change the UI for this. This way it's persistent without being intrusive. Particularly, it makes it clear if you screenshot a graph. If the project has no events from before Jan 1, 2015, the label will just say "All time" as before.




## New additional scope (metadata)

In addition to the above, I've piggy backed of my own PR (because we can use the same endpoint). Apologies in advance for the scope creep @macobo, this seemed like the fastest route and a good trade-off to the time I can spend in Engineering.

- The same `/api/projects/<id>/metadata` endpoint that sends information on the first event, now sends some additional metadata on the number of saved insights and number of custom dashboards (i.e. non-default ones) and we include those properties for the "project" group.
- The above is so that we can measure (among other things), the impact of having saved insights / dashboards to activation (see https://github.com/PostHog/product-internal/pull/261/files#r796622471 for context). CC @jamesefhawkins @marcushyett-ph

**Comment by clarkus** [comment]
image

How does a user reach out in this scenario? It could be frustrating to be told to reach out, but then offer no means of doing so. Not a huge deal, but if we really want people to engage with us on this we should provide a means of doing that via a link or some other action. 

I've mentioned this before in other reviews, but the absolute time ranges in the picker control are problematic. That control is for picking a range relative to the current. Show absolute values here makes it feel overloaded and hard to parse (to me at least). The chart, or corresponding table should represent the absolute range of time.

**Comment by paolodamico** [comment]
Re @clarkus,

- Re reaching out. I had originally made it so if you click on the tooltip icon, we open the help context menu at the top bar of the app, but this felt confusing. As interacting with the tooltip was impossible too, I've updated the copy to provide more help here.
- Re the "absolute date range". I disagree here. All time is an absolute date range in itself. Adding "Jan 1, 2015" only clarifies that data before that is not considered. I think that's particularly helpful when taking app screenshots. Thoughts?

**Comment by clarkus** [comment]
> Re the "absolute date range". I disagree here. All time is an absolute date range in itself. Adding "Jan 1, 2015" only clarifies that data before that is not considered. I think that's particularly helpful when taking app screenshots. Thoughts?

Agree to disagree. I don't think absolute values belong in the selection control. They increase the visual weight of a single item and break the pattern for a one off constraint. I genuinely think it makes the component harder to use and less scalable.  I am not optimizing for screenshots, I am optimizing for users being able to quickly and easily change the time range in an obvious way.

Why not communicate the 2015 cutoff in the tooltip itself? Do we need to show Jan 1/1/2015 every time? It just feels very heavy handed and like we're biasing the component towards an edge case.

**Comment by paolodamico** [comment]
@clarkus to clarify, the additional "Jan 1, 2015 - present" label only appears after you've selected "All time" (i.e. it does not appear in the dropdown list and only for projects with events before Jan 1, 2015 (in PostHog Cloud, these are only [69 projects](https://metabase.posthog.net/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJzZWxlY3QgY291bnQoZGlzdGluY3QgdGVhbV9pZCkgZnJvbSBldmVudHMgd2hlcmUgdGltZXN0YW1wIDwgJzIwMTUtMDEtMDEnOyIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjJ9LCJkaXNwbGF5IjoidGFibGUiLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=)). Further considering that only [~6% of insights are on all time](https://app.posthog.com/insights/i3b7dYEh/edit#filters=%7B%22insight%22%3A%22TRENDS%22%2C%22interval%22%3A%22week%22%2C%22display%22%3A%22ActionsLineGraph%22%2C%22actions%22%3A%5B%7B%22id%22%3A%2210784%22%2C%22name%22%3A%22Discoveries%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A0%7D%2C%7B%22id%22%3A%2210784%22%2C%22name%22%3A%22Discoveries%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A1%2C%22properties%22%3A%5B%7B%22key%22%3A%22date_from%22%2C%22value%22%3A%5B%22all%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D%7D%5D%2C%22events%22%3A%5B%5D%2C%22properties%22%3A%5B%5D%2C%22filter_test_accounts%22%3Atrue%2C%22new_entity%22%3A%5B%5D%2C%22date_from%22%3A%22-30d%22%2C%22date_to%22%3Anull%2C%22formula%22%3A%22B%2FA*100%22%7D), seems like this would seldom be shown. I advocate this because it provides clarity beyond a doubt, whether using PH or sharing screenshots (which we know is common). Is there another way we should display this but in a way that provides clarity even when sharing screenshots?

Still awaiting technical review anyways @mariusandra when you get a chance

**Review by mariusandra — CHANGES_REQUESTED** [review]
Facebook has ruined the word "meta" for me 😅 

I'm not sure if this is the way we should go though. 

First, UX wise, I agree with @clarkus - the knowledge of Jan 1 2015 is probably very irrelevant for me, even if someone stupidly ingested an event in 1970. So why show it? Add a `(i)` with a tooltip that gives this information, but no need to give your arbitrarily chosen date prime UX real-estate because Yuri from DevOps used a negative timestamp once in early 2022.

Second, isn't the solution not to "reach out", but "select an old date in the filter yourself"?

Third, why is there code in this PR to add the count of custom dashboards and saved insights onto the user's project? That seems like out of scope. 🤔 

Fourth, I'm not sure yet another metadata object and another loader for another set of data we'll need to mock (in addition to the app context, team, user, organisation members, instance status, group types, billing) is the right way to go. Some of the queries we'll want to add here probably shouldn't be calculated on the fly as well. Not against, but would appreciate some more reviews and engineering buy-in before agreeing to support this long term. 

Sorry for making this hard 😅

**Comment by paolodamico** [comment]
Thanks for the review @mariusandra! So looks like there's a lot of feedback against this selector thing, so I'll just leave the "All time" component as it is.

Re scope creep, I think it'd still be valuable to have that metadata for group analytics. The reason I went with the additional endpoint was because of @macobo's initial feedback, but that was based on other assumptions. Is it feasible to capture that metadata for group analytics and how would you recommend to do it?

File: posthog/api/team.py [hunk 2]
```diff
@@ -195,3 +193,46 @@ def reset_token(self, request: request.Request, id: str, **kwargs) -> response.R
...
+        except ImportError:
+            return None
+
+        try:
+            response = sync_execute(
+                "SELECT timestamp FROM events WHERE team_id = %(team_id)s ORDER BY timestamp LIMIT 1",
+                {"team_id": team.pk},
+            )[0][0]
+        except IndexError:
+            return None
```
[the remark this answers is not in the document]
Comment (paolodamico) [reply posthog/api/team.py:216]: "Yuri from DevOps" 🤣

**Comment by mariusandra** [comment]
I think an additional endpoint can make sense. We don't really have anything like this right now. It doesn't go in instance status. It could go on team or in the [frontend app context](https://github.com/PostHog/posthog/blob/a71e89960526701ecbdd01b32d3a209def0bb7b6/posthog/utils.py#L257) that already does some queries, but I fear we'll just add bloat to the app's startup time. Might be fine though, as then the data will always be there when the app loads...

However in this PR the logic reads more like "this project metadata endpoint returns an object with 3 properties, now we will parse out two of them and send them to posthog as project metadata". This adds another decision point that we could avoid --> pass the object directly, only return data that should be passed, etc.
